.github/workflows/ci_linting.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: Linters
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  linters:
```

Comment:
Are any of these steps dependent on each other? If not, I would think it would be better to have each of these as a separate job. We could do this with the `strategy.matrix` operator.

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

ex:

```yaml
jobs:
  lint:
    strategy:
      matrix:
        command:
          - |
            source ./codebuild/bin/s2n_setup_env.sh
            ./codebuild/bin/install_cppcheck.sh "$CPPCHECK_INSTALL_DIR"
            ./codebuild/bin/run_cppcheck.sh "$CPPCHECK_INSTALL_DIR"
          - |
            source ./codebuild/bin/s2n_setup_env.sh
            ./codebuild/bin/copyright_mistake_scanner.sh
          - |
            source ./codebuild/bin/s2n_setup_env.sh
            ./codebuild/bin/grep_simple_mistakes.sh
          - |
            source ./codebuild/bin/s2n_setup_env.sh
            ./codebuild/bin/run_kwstyle.sh
            ./codebuild/bin/cpp_style_comment_linter.sh
    steps:
      - uses: actions/checkout@v2
      - run: ${{ matrix.command }}
```

Reply:
Added.  The main issue I see with this approach is the lack of a friendly name for each (it doesn't appear that you can make the matrix structure very complex).  As a workaround, added an echo statement....